### PR TITLE
Adding Analytical rules to solution

### PR DESCRIPTION
**Pull request by jctaillandier**
Required items, please complete
   
   Change(s):
   - Adding a number of analytical rules to Solution deployment
   - Disabled rules by default to reduce noise
   - Also fixed a bug on solution deployment:
   
image002

   Reason for Change(s):
   - Most of our clients wanted those rules

   Version Updated:
   - Yes to 2.1.0

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
  

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by jctaillandier** [comment]
A few questions for Microsoft folks:

- How do I generate the ID field in first line of yaml?
- Can I run tests locally ? I have issues where clients deploying solution have an Error and I cannot reproduce locally...
- For the `Tactics` field, Is there a list of validated values I can use there ?

**Comment by v-spadarthi** [comment]
@jctaillandier : Validation is falining please fix it.

**Comment by jctaillandier** [comment]
> @jctaillandier : Validation is falining please fix it.

@v-spadarthi Can you look at my questions above?

**Comment by v-spadarthi** [comment]
@jctaillandier : Still you want to add anything in this PR if yes,Please chnage status Open to Draft
image
It is available in Rightside under Reviewers it will be available.

**Comment by jctaillandier** [comment]
Same as before, validation error does not explain or show where the error is

**Comment by v-spadarthi** [comment]
@jctaillandier : Please fix the validation errors 
image
If above snippet is not clear please click on Details for validation errors
image
image
image

**Comment by aprakash13** [comment]
> A few questions for Microsoft folks:
> 
> * How do I generate the ID field in first line of yaml?
> * Can I run tests locally ? I have issues where clients deploying solution have an Error and I cannot reproduce locally...
> * For the `Tactics` field, Is there a list of validated values I can use there ?

- ID is just a standard GUID. You can generate from just about any development tool, online GUID generator, or from PowerShell via the [New-GUID cmdlet](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-guid?view=powershell-6).

- There are some instructions here for running them locally:  https://github.com/Azure/Azure-Sentinel/blob/master/README.md#run-kql-validation-locally

- List of Valid Tactics:  https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/detectionTemplateSchemaValidation/Models/AttackTactic.cs

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+id: 467cbe7e-e6d4-4f4e-8f44-84dd01932c32
```
Comment (aprakash13) [inline Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml:1]: Please generate a new Id. the validations also seems to be having some errors here.
Error message:
Template Id: 467cbe7e-e6d4-4f4e-8f44-84dd01932c32 is not valid in Line: 1 col: 1

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+  - BruteForce
+relevantTechniques:
+  - T1207
+query: |
+  Firework_CL
```
Comment (aprakash13) [inline Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml:20]: There are some errors in the KQL Validation as the validator does not understand the table name. Can we please add a json that defines the schema of the Firework_CL table at the below location. The folder has examples of other custom tables. 
[image](https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables)

**Review by aprakash13 — CHANGES_REQUESTED** [review]
We need to fix the validation errors that are currently triggering. Added some suggestions for fixing those. Thanks for your contribution.

**Comment by v-spadarthi** [comment]
@jctaillandier: Could you please adress the @aprakash13  comments. Thanks

**Comment by v-spadarthi** [comment]
@jctaillandier: Could you please adress the @aprakash13 comments. Thanks

**Comment by v-atulyadav** [comment]
Hi @jctaillandier, please check response from @aprakash13 regarding validation error. Thanks

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml (hunk 2 above)
Comment (jctaillandier) [reply Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml:20]: The schema has hundreds of possible columns and depending on the type of data, the schema will differ. do I need to add all of them here ?

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml (hunk 1 above)
Comment (jctaillandier) [reply Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml]: Done

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - connectorId: Flare
+    dataTypes:
+      - Firework_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+  - BruteForce
```
[the remark this answers is not in the document]
Comment (jctaillandier) [reply Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml]: Done

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml (hunk 2 above)
Comment (jctaillandier) [reply Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml:19]: I added a file, but another error pops up saying my column names are not valid... 
```
  Failed Kqlvalidations.Tests.KqlValidationTests.Validate_DetectionQueries_HaveValidKql(fileName: "FlareCredentialLeaks.yaml", encodedFilePath: "L2hvbWUvdnN0cy93b3JrLzEvcy9Tb2x1dGlvbnMvRmxhcmUvQW"...) [113 ms]
  Error Message:
   Template Id: 9cb7c337-f170-4af6-b0e8-b6b7552d762d is not valid in Line: 2 col: 40
                    Errors: The name 'source_s' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '51..59'
```

**Comment by jctaillandier** [comment]
Everything seems pretty good with validation checks. Only issue left is on ARM-ttk validations
```
#17 5.246     [-] Variables Must Be Referenced (13 ms)
#17 5.246         Unreferenced variable: playbook1-AzureSentinelConnectionName
#17 5.246         Unreferenced variable: playbook1-o365ConnectionName
```
Does it have anything to do with my changes?

**Review by aprakash13 — APPROVED** [review]
Looks good from an analytic perspective. 
Please feel free to merge if other components look good.

**Comment by jctaillandier** [comment]
Do I have anything else to do ?

**Comment by jctaillandier** [comment]
@v-spadarthi Can we merge this ?

**Comment by v-atulyadav** [comment]
Looking into validations

**Comment by v-sabiraj** [comment]
@jctaillandier, please update the branch from master, thanks.

**Comment by jctaillandier** [comment]
@v-sabiraj Just did, same ttk error
```
#13 5.054   Variables Must Be Referenced
#13 5.054     [-] Variables Must Be Referenced (13 ms)
#13 5.054         Unreferenced variable: playbook1-AzureSentinelConnectionName
#13 5.054         Unreferenced variable: playbook1-o365ConnectionName
```

**Comment by v-atulyadav** [comment]
Hi @jctaillandier, we will check this validation failure and provide you with feedback as soon as possible. Thanks.

**Comment by v-atulyadav** [comment]
Hi @jctaillandier , please modify the solution input file and also in Data folder as well for Flare solution (Add Analytics in solution input file) and version change 2.0.2 and please create package again. Thanks

**Comment by jctaillandier** [comment]
@v-atulyadav Tried it, and a few other things, still can't make it pass.

Seems an issue with variables naming and references. Now about values set for `id`

**Comment by v-atulyadav** [comment]
Hi @jctaillandier, I would appreciate if you could take the latest from master and update the branch from master and check it. Thanks.

**Comment by v-atulyadav** [comment]
Hi @jctaillandier, Take the latest of master branch and pull your PR branch, then click on update from master and push the changes. 

image

**Comment by jctaillandier** [comment]
@v-atulyadav 
 I had already done, that. Just did it again.

**Comment by v-sabiraj** [comment]
Hi @jctaillandier, the branch seems to be forked, you need to take update from Azure:master to Flared:jct/add_analytics_rules, thanks.

**Pull request closed without merging**